game-app/game-core/src/main/java/games/strategy/engine/data/RuleComparator.java

```diff
@@ -0,0 +1,52 @@
+package games.strategy.engine.data;
+
+import games.strategy.triplea.attachments.UnitTypeComparator;
+import java.util.Comparator;
+
+public class RuleComparator<T extends Rule> implements Comparator<T> {
+  private static final UnitTypeComparator unitTypeComparator = new UnitTypeComparator();
```

Comment:
I would suggest to not make this static. If the unit type comparator makes use of any caching that assumes the underlying units don't change, then having a static instance can lead to bugs.

Reply:
> I would suggest to not make this static. If the unit type comparator makes use of any caching that assumes the underlying units don't change, then having a static instance can lead to bugs.

Haven't thought about that. I've been learning a lot here. 😁
Thank you.